src/Nethermind/Nethermind.Logging/PathUtils.cs

```diff
@@ -25,7 +28,17 @@ public static class PathUtils
     {
         public static string GetExecutingDirectory()
         {
-            return Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);;
+            var process = Process.GetCurrentProcess();
+            if (process.ProcessName.StartsWith("dotnet", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            }
+            
+            var filePath = process.MainModule.FileName;
+            var fileName = filePath.Split(Path.DirectorySeparatorChar).LastOrDefault();
+            var fileNameIndex = filePath.LastIndexOf(fileName, StringComparison.InvariantCultureIgnoreCase);
+
+            return filePath.Substring(0, fileNameIndex);
```

Comment:
Isn't it the same as https://docs.microsoft.com/pl-pl/dotnet/api/system.io.path.getdirectoryname?view=netcore-3.0 ?